mne_bids_pipeline/steps/init/_01_init_derivatives_dir.py

```diff
@@ -10,16 +10,14 @@
 
 from ..._config_utils import _bids_kwargs, get_sessions, get_subjects
 from ..._logging import gen_log_kwargs, logger
-from ..._run import failsafe_run
+from ..._run import _prep_out_files, failsafe_run
 
 
-def init_dataset(cfg) -> None:
+@failsafe_run()
+def init_dataset(cfg: SimpleNamespace, exec_params: SimpleNamespace) -> dict:
```

Comment:
Is it possible to specify the types of the `dict` contents? `dict[str, Any]` in case the type of the values is difficult to capture (I didn't look into the details of that code)